sdk/couchbase/src/bucket_settings.rs

```diff
@@ -0,0 +1,478 @@
+                EvictionPolicyType::VALUE_ONLY
+            }
+            couchbase_core::mgmtx::bucket_settings::EvictionPolicyType::FULL_EVICTION => {
+                EvictionPolicyType::FULL_EVICTION
+            }
+            couchbase_core::mgmtx::bucket_settings::EvictionPolicyType::NOT_RECENTLY_USED => {
+                EvictionPolicyType::NOT_RECENTLY_USED
+            }
+            couchbase_core::mgmtx::bucket_settings::EvictionPolicyType::NO_EVICTION => {
+                EvictionPolicyType::NO_EVICTION
+            }
+            _ => EvictionPolicyType::unknown(value.to_string()),
+        }
+    }
+}
+
+impl From<couchbase_core::mgmtx::bucket_settings::CompressionMode> for CompressionMode {
+    fn from(value: couchbase_core::mgmtx::bucket_settings::CompressionMode) -> Self {
+        match value {
+            couchbase_core::mgmtx::bucket_settings::CompressionMode::OFF => CompressionMode::OFF,
+            couchbase_core::mgmtx::bucket_settings::CompressionMode::PASSIVE => {
+                CompressionMode::PASSIVE
+            }
+            couchbase_core::mgmtx::bucket_settings::CompressionMode::ACTIVE => {
+                CompressionMode::ACTIVE
+            }
+            _ => CompressionMode::unknown(value.to_string()),
+        }
+    }
+}
+
+impl From<couchbase_core::mgmtx::bucket_settings::ConflictResolutionType>
+    for ConflictResolutionType
+{
+    fn from(value: couchbase_core::mgmtx::bucket_settings::ConflictResolutionType) -> Self {
+        match value {
+            couchbase_core::mgmtx::bucket_settings::ConflictResolutionType::SEQUENCE_NUMBER => {
+                ConflictResolutionType::SEQUENCE_NUMBER
+            }
+            couchbase_core::mgmtx::bucket_settings::ConflictResolutionType::TIMESTAMP => {
+                ConflictResolutionType::TIMESTAMP
+            }
+            couchbase_core::mgmtx::bucket_settings::ConflictResolutionType::CUSTOM => {
+                ConflictResolutionType::CUSTOM
+            }
+            _ => ConflictResolutionType::unknown(value.to_string()),
+        }
+    }
+}
+
+impl From<couchbase_core::mgmtx::bucket_settings::StorageBackend> for StorageBackend {
+    fn from(value: couchbase_core::mgmtx::bucket_settings::StorageBackend) -> Self {
+        match value {
+            couchbase_core::mgmtx::bucket_settings::StorageBackend::COUCHSTORE => {
+                StorageBackend::COUCHSTORE
+            }
+            couchbase_core::mgmtx::bucket_settings::StorageBackend::MAGMA => StorageBackend::MAGMA,
+            _ => StorageBackend::unknown(value.to_string()),
+        }
+    }
+}
+
+impl From<couchbase_core::mgmtx::bucket_settings::BucketDef> for BucketSettings {
+    fn from(value: BucketDef) -> Self {
+        Self {
+            name: value.name,
+            ram_quota_mb: value
+                .bucket_settings
+                .mutable_bucket_settings
+                .ram_quota_mb
+                .unwrap_or(0),
```

Comment:
Why is a default of 0 being used? We don't seem to do something like that for any other settings?

Reply:
As far as I'm aware this can't actually be empty, but we use the same type for create and get so it has to be optional at the core level but we don't allow it to be optional in the public API.

Reply:
That's right, for create or update it just leaves it up to server to figure out what it wants to do.